pkg/envoy/proxy.go

```diff
@@ -275,3 +282,14 @@ func (p *Proxy) buildCommand(ctx context.Context, cfgPath string) *exec.Cmd {
 
 	return cmd
 }
+
+// RemoveArgAndGetValue Function to get new args after removing given key
+// and also returns the value of key
+func RemoveArgAndGetValue(stringAr []string, key string) ([]string, string) {
```

Comment:
Should this be a private function instead?

Reply:
ok yes, can be private